caddytls/sync_locker.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2015 Light Code Labs, LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caddytls
+
+import (
+	"fmt"
+	"sync"
+)
+
+var _ Locker = &syncLock{}
+
+type syncLock struct {
+	nameLocks   map[string]*sync.WaitGroup
+	nameLocksMu sync.Mutex
+}
+
+// Lock attempts to acquire the lock and blocks until
+// the lock can be acquired.
+func (s *syncLock) Lock(name string) error {
+	for {
+		w, err := s.TryLock(name)
+		if err != nil {
+			return err
+		}
+
+		if w == nil {
+			return nil
+		}
+
+		w.Wait()
+	}
+}
```

Comment:
I'm not opposed to the Lock method, but I want to make sure I understand its value over TryLock. I mean, it makes sense to have one, but in implementing the locking initially in caddytls, I never used a plain Lock; I never wanted to block until a lock was obtained.

Reply:
@mholt there is no reason to have `Lock` right now but I imagine a couple usages when using in a distributed environment. It is alright for me to remove it for now and bring it back later when it is needed.

Reply:
Okay, sounds good. Let's remove it unless it's needed. We can add it later.

Reply:
@mholt Done.